services/column.service.go

```diff
@@ -0,0 +1,112 @@
+package services
+
+import (
+	"fmt"
+	"github.com/evsyukovmv/taskmanager/models"
+	"github.com/go-playground/validator/v10"
+	"sync"
+)
+
+type ColumnService struct {
+	storage  ColumnStorage
+	validator *validator.Validate
+}
+
+var columnOnce sync.Once
+var columnSingleton *ColumnService
+
+func NewColumnService(cs ColumnStorage) {
+	columnOnce.Do(func() {
+		columnSingleton = &ColumnService{storage: cs, validator: validator.New()}
+	})
+}
+
+func ForColumn() *ColumnService {
+	return columnSingleton
+}
+
+func (s *ColumnService) validate(column *models.Column) error {
+	return s.validator.Struct(column)
+}
+
+func (s *ColumnService) Create(c *models.Column) error {
+	err := s.validate(c)
+	if err != nil {
+		return err
+	}
+
+	return s.storage.Create(c)
+}
+
+func (s *ColumnService) Delete(columnId int) (*models.Column, error) {
+	c, err := s.storage.GetById(columnId)
+	if err != nil {
+		return c, err
+	}
+
+	count, err := s.storage.CountInProject(c.ProjectId)
+	if err != nil {
+		return c, err
+	}
+
+	if  count < 2 {
+		return c, fmt.Errorf("deleting the last column is not allowed")
```

Comment:
use `errors.New()` because there is no formatting options passed to `fmt.Errorf`